core/src/main/java/org/elasticsearch/search/internal/InternalSearchHit.java

```diff
@@ -342,6 +363,10 @@ public void sortValues(Object[] sortValues, DocValueFormat[] sortValueFormats) {
         this.sortValues = new SearchSortValues(sortValues, sortValueFormats);
     }
 
+    void sortValues(SearchSortValues sortValues) {
```

Comment:
nit: shall we make this the only setter (make it public also) and have caller of the other setter for the same member call this instead?